src/app/view/sharing/SharingScreen.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react'
+import { View, Text, StyleSheet } from 'react-native'
+
+import { useSharingMode } from './useSharing'
+
+export const SharingScreen = (): JSX.Element => {
+  const [numberOfFiles, setNumberOfFiles] = useState(0)
+  const { filesToUpload } = useSharingMode()
+
+  useEffect(() => {
+    setNumberOfFiles(filesToUpload.length)
+
+    return () => {
+      setNumberOfFiles(0)
+    }
+  }, [filesToUpload])
```

Comment:
Why do we need this effect? Refering `filesToUpload.length` directly in the jsx code seems enough?

Reply:
I guess yes, my only issue is calling the getter too early for instance.